packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -333,21 +367,119 @@ const DiagramContent: React.FunctionComponent<{
     [onAddFieldToObjectField]
   );
 
-  const diagramProps = useMemo(
-    () => ({
-      isDarkMode,
-      title: diagramLabel,
-      edges,
-      nodes,
-      onAddFieldToNodeClick: onClickAddFieldToCollection,
-      onAddFieldToObjectFieldClick: onClickAddFieldToObjectField,
-      onNodeClick,
-      onPaneClick,
-      onEdgeClick,
-      onFieldClick,
-      onNodeDragStop,
-      onConnect,
-    }),
+  const onSelectionChange = useCallback(
+    ({ nodes, edges }: { nodes: NodeProps[]; edges: EdgeProps[] }) => {
+      // Select the first selected item (if any) in the diagram
+      // since unlike react-flow, we only support single item selections
+      if (nodes.length > 0) {
+        onNodeClick(null, nodes[0]);
+        return;
+      }
+      if (edges.length > 0) {
+        onEdgeClick(null, edges[0]);
+        return;
+      }
+      // If nothing is selected, clear the selection
+      onDiagramBackgroundClicked(true);
+    },
+    [onDiagramBackgroundClicked, onNodeClick, onEdgeClick]
+  );
+
+  const onKeyDown = useCallback(
+    (event: React.KeyboardEvent | KeyboardEvent) => {
+      // If no element is focused, some inputs should still be treated as targeting the diagram
+      // (e.g. undo/redo)
+      const isUnfocusedEvent =
+        event.target === document.body ||
+        event.target === document.documentElement;
```

Comment:
I think the alternative here would be focusing the diagram itself when blurring away from a selection, but that seems like it would be harder to implement and more overall effort